tests/cpp/test_serial_gridreduce.cpp

```diff
@@ -37,110 +37,91 @@ using SerialGridReductionTest = NVFuserTest;
 
 TEST_F(SerialGridReductionTest, Scheduling) {
   for (bool serial : {true, false}) {
-    // For serial grid reduction we test that re-using semaphore buffers works
-    // properly. This will cause a failure if serial grid reduction does not
-    // properly clean up its semaphores.
-    std::vector<bool> reuse_zeroed_memory_values{false};
-    if (serial) {
-      reuse_zeroed_memory_values.push_back(true);
-    }
-    for (bool reuse_zeroed_memory : reuse_zeroed_memory_values) {
-      EnableOptionsGuard eog;
-      if (reuse_zeroed_memory) {
-        EnableOptionsGuard::getCurOptions().set(
-            EnableOption::ReuseZeroedMemory);
-      } else {
-        EnableOptionsGuard::getCurOptions().unset(
-            EnableOption::ReuseZeroedMemory);
-      }
```

Comment:
This change just removes any explicit mention of zeroed memory. Using `NVFUSER_DUMP=global_zeroed_memory`, we can confirm that the zeroed memory pool was used.

Reply:
```
# ninja -C build -j10 nvfuser_tests && NVFUSER_DUMP=global_zeroed_memory build/nvfuser_tests --gtest_filter=SerialGridReductionTest.Scheduling
ninja: Entering directory `build'
ninja: no work to do.
Running main() from /opt/pytorch/nvfuser/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = SerialGridReductionTest.Scheduling
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from SerialGridReductionTest
[ RUN      ] SerialGridReductionTest.Scheduling
[global zeroed memory] Resizing arena to 512 bytes
[global zeroed memory] Allocating byte range: 0 to 512 bytes
[global zeroed memory] Resetting allocated bytes to 0
[global zeroed memory] Allocating byte range: 0 to 512 bytes
[global zeroed memory] Resetting allocated bytes to 0
[global zeroed memory] Resetting allocated bytes to 0
[global zeroed memory] Resetting allocated bytes to 0
[       OK ] SerialGridReductionTest.Scheduling (7429 ms)
[----------] 1 test from SerialGridReductionTest (7429 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (7429 ms total)
[  PASSED  ] 1 test.
```